OsmAndMapCreator/src/net/osmand/data/preparation/IndexPoiCreator.java

```diff
@@ -155,6 +156,17 @@ public void iterateEntity(Entity e, OsmDbAccessorContext ctx, boolean basemap) t
 		}
 	}
 
+	public static long generateRelationId(Amenity amenity) {
+		LatLon loc = amenity.getLocation();
+		return -((long) MapUtils.get31TileNumberX(loc.getLongitude()) << 31 | (long) MapUtils.get31TileNumberY(loc.getLatitude()));
```

Comment:
This is a bad idea, in that case we store huge numbers 64 bits without a reasonable cause, they simply could be decremental from -100.